mmv1/third_party/terraform/website/docs/d/monitoring_istio_canonical_service.html.markdown

```diff
@@ -0,0 +1,84 @@
+---
+subcategory: "Cloud (Stackdriver) Monitoring"
+layout: "google"
+page_title: "Google: google_monitoring_istio_canonical_service"
+sidebar_current: "docs-google-datasource-monitoring-istio-canonical-service"
+description: |-
+  An Monitoring Service resource created automatically by GCP to monitor an
+  Istio Canonical service.
+---
+
+# google\_monitoring\_istio\_canonical\_service
+
+A Monitoring Service is the root resource under which operational aspects of a
+generic service are accessible. A service is some discrete, autonomous, and
+network-accessible unit, designed to solve an individual concern
+
+A monitoring Istio Canonical Service is automatically created by GCP to monitor
+Istio Canonical Services.
+
+
+To get more information about Service, see:
+
+* [API documentation](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/services)
+* How-to Guides
+    * [Service Monitoring](https://cloud.google.com/monitoring/service-monitoring)
+    * [Monitoring API Documentation](https://cloud.google.com/monitoring/api/v3/)
+
+## Example Usage - Monitoring Istio Canonical Service
+
+
+```hcl
+# Monitors the default MeshIstio service
+data "google_monitoring_istio_canonical_service" "default" {
+        mesh_uid = "proj-573164786102"
+        canonical_service_namespace = "istio-system" 
+        canonical_service = "prometheus"
+}
+```
+
+## Argument Reference
+
+The arguments of this data source act as filters for identifying a given -created service.
+
+The given filters must match exactly one service whose data will be exported as attributes. The following arguments are supported:
+
+The following fields must be specified:
+
+* `mesh_uid` - (Required) Identifier for the mesh in which this Istio service is defined.
+  Corresponds to the meshUid metric label in Istio metrics.
+
+* `canonical_service_namespace` - (Required) The namespace of the canonical service underlying this service.
+  Corresponds to the destination_canonical_service_namespace metric label in Istio metrics.
+
+* `canonical_service` - (Required) The name of the canonical service underlying this service.
+  Corresponds to the destination_canonical_service_name metric label in label in Istio metrics.
+  
+- - -
+
+Other optional fields include:
+
+* `project` - (Optional) The ID of the project in which the resource belongs.
+    If it is not provided, the provider project is used.
+
+## Attributes Reference
+
+In addition to the arguments listed above, the following computed attributes are exported:
+
+* `name` -
+  The full REST resource name for this channel. The syntax is:
+  `projects/[PROJECT_ID]/services/[SERVICE_ID]`.
+
+* `display_name` -
+  Name used for UI elements listing this (Monitoring) Service.
+
+* `telemetry` -
```

Comment:
Does this work? I don't see anywhere that we read this value or these other attributes from the API response, so I can't imagine that it would ever be populated. They also aren't included in the schema for the datasource, so there should be no possible way for this to work

Reply:
All return populated except for "telemetry":

I tested with a real GKE with ASM implementation on a test project:

main.tf
```
data "google_monitoring_istio_canonical_service" "default" {
  mesh_uid = "proj-57221930739"
  canonical_service_namespace = "default"
  canonical_service = "helloserver"
}

output "test_google_monitoring_istio_canonical_service" {
  value = data.google_monitoring_istio_canonical_service.default
}
```

Output:
```
Outputs:

test_google_monitoring_istio_canonical_service = {
  "canonical_service" = "helloserver"
  "canonical_service_namespace" = "default"
  "display_name" = "helloserver"
  "id" = "projects/57221930739/services/canonical-ist:proj-57221930739-default-helloserver"
  "mesh_uid" = "proj-57221930739"
  "name" = "projects/57221930739/services/canonical-ist:proj-57221930739-default-helloserver"
  "project" = "franzramirez-support-workspace"
  "service_id" = "canonical-ist:proj-57221930739-default-helloserver"
  "telemetry" = tolist([])
}
```